src/PetriEngine/Colored/EvaluationVisitor.cpp

```diff
@@ -129,8 +129,6 @@ namespace PetriEngine {
             (*sub)[0]->visit(*this);
             auto lhs = _mres;
             (*sub)[1]->visit(*this);
-            if (!_mres.isSubsetOf(lhs))
-                throw base_error("RHS of subtraction is not a subset of LHS");
```

Comment:
Is this OK? I'd almost assume that this throw exist for a reason.

Reply:
It seems that the check is to avoid redundant subtractions. E.g. in the negative arc model in the bug report 1'(1,1) - 1'(4,2) is equivalent to just 1'(1,1), but this will fail with the check on, whereas without it will do the subtraction operation the result would end up being just 1'(1,1).